hicommonwealth/commonwealth
# [Sign-in Revamp]: EVM wallets submodal implementation in Auth Modal

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6387

## Description of Changes
- Adds new sub-modal in AuthModal for EVM wallets (wallet connect, coinbase, metamask) + their integrations
- Adds community-specific wallet integrations (as it was in old modal)
  - In 'Axie-Infinity', only `Ronin` wallet option is available
  - In 'Terra', only `StationWallet` and `Terra-WalletConnect` (wallet connect for terra) options are available
  - In 'Near', only `Near` wallet option is available
  - In 'Evmos' / 'Injective', show `cosm-metamask` (metamask for cosmos, it has the same button as metamask wallet, but it works for cosmos community) and `keplr-ethereum` wallets (it has the same button as Keplr wallet)

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_NEW_SIGN_IN_MODAL` in `.env`
- Install and setup EVM wallets (metamask/coinbase).
- Logout and visit any non-community page or any Ethereum-based community.
- Open the AuthModal, and click on the `Wallet connect` option to open the EVM wallets submodal.
- From the list in submodal, click on any wallet (`Coinbase` might not work in local) and try to login, verify it logs you in correctly and closes both modals.
- Logout and try to login with `WalletConnect` (you might have to install some wallet on mobile (if testing on mobile), i tried with `Zerion`) from the evm wallets submodal, while it is logging in
  - verify you see the reset wallet connect button (it will not work in local)
  - verify it logs you in correctly.
<img width="461" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/51641047/64169939-ccd5-46e0-aefb-5e939ed517b8">

- Visit the `/axie-infinity` community, open the auth modal and verify you see the `Ronin` wallet option if you have that wallet installed, else verify you see "No wallets available" message. Verify that logging in with that wallets works.
- Visit the `/terra` community, open the auth modal and verify you see the `StationWallet` wallet option if you have that wallet installed and you see the 'WalletConnect (Terra)' option (this is different from the general wallet connect flow), else verify you see "No wallets available" message. Verify that logging in with these wallets works.
- Visit the `/near` community, open the auth modal and verify you see the `Near` wallet option. Verify that logging in with this wallet works.

## Deployment Plan
Part of https://github.com/hicommonwealth/commonwealth/milestone/38

## Other Considerations
`FLAG_NEW_SIGN_IN_MODAL` should not be enabled in prod yet.
